datafusion/functions/src/datetime/to_local_time.rs

```diff
@@ -0,0 +1,601 @@
+                            // if no timezone specificed, just return the input
+                            Ok(ColumnarValue::Array(Arc::new(array)))
+                        }
+                        Timestamp(Nanosecond, Some(tz)) => {
+                            transform_array::<TimestampNanosecondType>(&array, tz)
+                        }
+                        Timestamp(Microsecond, Some(tz)) => {
+                            transform_array::<TimestampMicrosecondType>(&array, tz)
+                        }
+                        Timestamp(Millisecond, Some(tz)) => {
+                            transform_array::<TimestampMillisecondType>(&array, tz)
+                        }
+                        Timestamp(Second, Some(tz)) => {
+                            transform_array::<TimestampSecondType>(&array, tz)
+                        }
+                        _ => {
+                            exec_err!("to_local_time function requires timestamp argument in array, got {:?}", array.data_type())
+                        }
+                    }
+                }
+                _ => {
+                    exec_err!(
+                        "to_local_time function requires timestamp argument, got {:?}",
+                        time_value.data_type()
+                    )
+                }
+            },
+            _ => {
+                exec_err!(
+                    "to_local_time function requires timestamp argument, got {:?}",
+                    arg_type
+                )
+            }
+        }
+    }
+}
+
+/// This function converts a timestamp with a timezone to a timestamp without a timezone.
+/// The display value of the adjusted timestamp remain the same, but the underlying timestamp
+/// representation is adjusted according to the relative timezone offset to UTC.
+///
+/// This function uses chrono to handle daylight saving time changes.
+///
+/// For example,
+///
+/// ```text
+/// '2019-03-31T01:00:00Z'::timestamp at time zone 'Europe/Brussels'
+/// ```
+///
+/// is displayed as follows in datafusion-cli:
+///
+/// ```text
+/// 2019-03-31T01:00:00+01:00
+/// ```
+///
+/// and is represented in DataFusion as:
+///
+/// ```text
+/// TimestampNanosecond(Some(1_553_990_400_000_000_000), Some("Europe/Brussels"))
+/// ```
+///
+/// To strip off the timezone while keeping the display value the same, we need to
+/// adjust the underlying timestamp with the timezone offset value using `adjust_to_local_time()`
+///
+/// ```text
+/// adjust_to_local_time(1_553_990_400_000_000_000, "Europe/Brussels") --> 1_553_994_000_000_000_000
+/// ```
+///
+/// The difference between `1_553_990_400_000_000_000` and `1_553_994_000_000_000_000` is
+/// `3600_000_000_000` ns, which corresponds to 1 hour. This matches with the timezone
+/// offset for "Europe/Brussels". Consequently, DataFusion can represent the timestamp in
+/// local time (with no offset or timezone information) as
```

Comment:
It might also be useful to mention here that the offset is not constant, as it varies based on daylight savings time  (which makes this so tricky)

Something like

```suggestion
/// The difference between `1_553_990_400_000_000_000` and `1_553_994_000_000_000_000` is
/// `3600_000_000_000` ns, which corresponds to 1 hour. This matches with the timezone
/// offset for "Europe/Brussels" for this date (the offset varies with daylight savings time).
/// Consequently, DataFusion can represent the timestamp in
/// local time (with no offset or timezone information) as
```

Reply:
Good point to mention daylight saving here! Updated in https://github.com/apache/datafusion/pull/11347/commits/ae9f1ad38668d3f080b59f15b7653828861c3bae